### PR TITLE
atmel-samd: Enable 8-bit audio recording support even though it'll be quiet.

### LINKED ISSUE #226: Example for audiobusio.PDMIn doesn't work
When using the examples here: https://circuitpython.readthedocs.io/en/latest/shared-bindings/audiobusio/PDMIn.html I get the following error using RC1 of CircuitPython 2.0:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid clock pin
```

I notice that the argument order for instantiating the `PDMIn` class has changed (`MICROPHONE_CLOCK` used to come first). Can you confirm that the signature for `__init__` is not going to change further?

### PR DESCRIPTION
**Pull request by tannewt**
Also update the examples.

Fixes #226

### CODE REVIEW

File: shared-bindings/audiobusio/PDMIn.c [hunk 1]
```diff
@@ -53,14 +53,28 @@
...
+//|   Record 16-bit unsigned samples to buffer::
+//|
+//|     import audiobusio
+//|     import board
+//|
+//|     # Prep a buffer to record into. The array interface doesn't allow for
+//|     # constructing with a set size so we append to it until we have the size
+//|     # we want.
+//|     b = array.array("H")
+//|     for i in range(200):
```
Comment (dhalbert) [inline shared-bindings/audiobusio/PDMIn.c]: Supposedly faster is `b = array.array("H", [0] * 200)`. It should generate less garbage.

`array.array("H", [0]) * 200` is supposed to be even faster but doesn't work in CPy.

https://stackoverflow.com/questions/3214288/what-is-the-fastest-way-to-initialize-an-integer-array-in-python

Comment (tannewt) [reply shared-bindings/audiobusio/PDMIn.c]: I'm more worried about memory impact than speed. Doing the [0] * 200 inside first creates a list that is then converted to an array.

Comment (dhalbert) [reply shared-bindings/audiobusio/PDMIn.c]: Yah, but will appending to an `array.array` create a new array of size+1? I thought it was contiguous memory.

Comment (tannewt) [reply shared-bindings/audiobusio/PDMIn.c]: Looks like its len+8 items each time using m_renew which will try to grow in place I believe. https://github.com/adafruit/circuitpython/blob/master/py/objarray.c#L319

Long term we probably want to switch to a multiply of the array `array.array("H", [0]) * 200` but its not supported now.

Comment (dhalbert) [reply shared-bindings/audiobusio/PDMIn.c]: OK, so never mind for now if you want.

Comment (tannewt) [reply shared-bindings/audiobusio/PDMIn.c]: Documented in #236

**Pull request merged**